glean-core/src/ac_migration/mod.rs

```diff
@@ -0,0 +1,108 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+//! A module containing glean-core code for supporting data migration
+//! from glean-ac.
+
+use crate::util::truncate_string_at_boundary;
+use std::collections::HashMap;
+
+use super::Glean;
+use super::PingMaker;
+
+const GLEAN_AC_SEQUENCE_SUFFIX: &str = "_seq";
+
+/// The Glean A-C metadata migration object.
+#[derive(Debug, Clone)]
+pub struct ACMetadata {
+    /// A map in which entries are (`<ping_name>_seq`, nextSequenceNumber)..
+    pub sequence_numbers: HashMap<String, i32>
+}
+
+/// Stores the sequence numbers from glean-ac in glean-core.
+pub(super) fn migrate_sequence_numbers(
+    glean: &Glean,
+    seq_numbers: HashMap<String, i32>
+) {
+    let ping_maker = PingMaker::new();
+
+    for (store_name_with_suffix, next_seq) in seq_numbers.iter() {
```

Comment:
We own `seq_numbers`, we can iterate and consume it with `.into_iter()` and save us the `&0` below.